egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -139,7 +149,48 @@ def AddConvolutionLayer(config_lines, name, input,
     num_y_steps = (1 + (input_y_dim - filt_y_dim) / filt_y_step)
     output_dim = num_x_steps * num_y_steps * num_filters;
     return {'descriptor':  '{0}_conv_t'.format(name),
-            'dimension': output_dim}
+            'dimension': output_dim,
+            '3d-dim': [num_x_steps, num_y_steps, num_filters],
+            'vectorization': 'zyx'}
+
+
+def AddMaxpoolingLayer(config_lines, name, input,
+                      input_x_dim, input_y_dim, input_z_dim,
```

Comment:
@vijayaditya , Are you talking about the AddConvolutionLayer() ? The assert statement is already there.